### PR TITLE
download module

### PR DESCRIPTION
**Pull request by namila007**
Signed-off-by: Namila Bandara <namila007@gmail.com>

# PR for issue #

### CODE REVIEW

**Comment by namila007** [comment]
issue #6  #7

File: src/main/java/com/octopus/Downloader/Downloader.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            this.filename = threadsize + url.substring(url.lastIndexOf("/") + 1);
+            System.out.println(filename);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    private void downloadFile() {
+
```
Comment (kasvith) [inline src/main/java/com/octopus/Downloader/Downloader.java]: We need to use JavaFX way of threading

File: src/main/java/com/octopus/Downloader/DownloadWorker.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/*
```
Comment (kasvith) [inline src/main/java/com/octopus/Downloader/DownloadWorker.java]: Package names should not have PascalCase

File: src/main/java/com/octopus/Downloader/HttpDownloadWorker.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+     * @param end
+     * @param filename
+     */
+    public HttpDownloadWorker(URL url, int start, int end, String filename) {
+        super(url, start, end, filename);
+    }
+
+
+    @Override
+    public void run() {
```
Comment (kasvith) [inline src/main/java/com/octopus/Downloader/HttpDownloadWorker.java]: This method cannot be used

**Comment by kasvith** [comment]
for now i will merge this

**Pull request merged**